packages/ui-modal/src/Modal/index.tsx

```diff
@@ -160,23 +149,14 @@ class Modal extends Component<ModalProps, ModalState> {
   renderChildren() {
     const { children, variant, overflow } = this.props
 
-    return Children.map(
-      children as (HeaderChild | BodyChild | FooterChild)[],
-      (child) => {
-        if (!child) return // ignore null, falsy children
-
-        if (matchComponentTypes<BodyChild>(child, [ModalBody])) {
-          return safeCloneElement(child, {
-            variant: variant,
-            overflow: child.props.overflow || overflow
-          })
-        } else {
-          return safeCloneElement(child, {
-            variant: variant
-          })
-        }
-      }
-    )
+    return Children.map(children, (child: any) => {
+      if (!child) return // ignore null, falsy children
+
+      return safeCloneElement(child, {
+        variant: variant,
+        overflow: child?.props?.overflow || overflow
+      })
```

Comment:
This line is causing hard to debug issues in some edge cases. If you add a `Tabs` or an `Alert` as a child you will get an exception because we use the `variant` prop often but not with the `inverse` value. AFAIK we cant do much here, I would just mention it in the docs that these 2 props are passed down to the children